amp/WEB-INF/src/org/digijava/kernel/validation/ConstraintDescriptors.java

```diff
@@ -0,0 +1,31 @@
+package org.digijava.kernel.validation;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+
+import org.apache.commons.collections.CollectionUtils;
+
+/**
+ * Container for constraint descriptors.
+ *
+ * @author Octavian Ciubotaru
+ */
+public class ConstraintDescriptors {
+
+    private List<ConstraintDescriptor> descriptors;
+
+    public ConstraintDescriptors(List<ConstraintDescriptor> descriptors) {
+        this.descriptors = descriptors;
+    }
+
+    public List<ConstraintDescriptor> getDescriptorsFor(Set<Class<?>> groups) {
+        ArrayList<ConstraintDescriptor> result = new ArrayList<>();
+        for (ConstraintDescriptor descriptor : descriptors) {
+            if (descriptor.getGroups().isEmpty() || CollectionUtils.containsAny(descriptor.getGroups(), groups)) {
```

Comment:
Groups are additive. It is impossible to disable or skip validators from Default group.

Reply:
Hm, not clear how exactly. Can you please provide more concrete explanation?

I am trying to guess myself how exactly we'll achieve it. As I can see there is no concrete Default.class group. So I suppose when a `descriptor.getGroups().isEmpty()`, then the validator is declared for the default group. Is my guessing right?

On the other hand we can specify for `InterchangeableValidator.group` let's say `API.class`. If we pass empty `groups` here (which per comment means pick all for default group), then code wise it won't pick the constraints with API.class, means they won't be additive on top of Default group and can be skipped from Default group in fact.

Reply:
You're guessing correctly, when groups is empty it means default group.
If we pass to this method empty `groups` then it is correct that it doesn't pick constraints for API group. If one wants to validate for API group, the parameter to `getDescriptorsFor` must be a singleton set with API group.

Reply:
Added java docs to clarify this case.

Reply:
Thanks for java docs. Sure, it makes total sense to me that when `API.class` set at an `InterchangeableValidator.group` means it will be skipped during the Default group validation (when `groups` is empty in `getDescriptorsFor`). It sounded opposite when you explained that  it is impossibile to skip validators from Default group and that groups are additive. However looks like setting any group explicitly means the constraint will be skipped during the Default group.